commerce_coordinator/apps/titan/pipeline.py

```diff
@@ -100,6 +100,45 @@ def run_filter(self, edx_lms_user_id, payment_number=None, order_uuid=None, **kw
         }
 
 
+class ValidateOrderReadyForDraftPayment(PipelineStep):
+    """
+    Stops the pipeline if the current order is not eligible for a draft payment.
+
+    An order is not eligible for a draft payment when it has already been paid
+    or is currently processing.
+    """
+
+    def run_filter(self, order_data, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Execute a filter with the signature specified.
+
+        Args:
+            order_data (dict): The active order for the user.
+            kwargs['recent_payment'] (dict): The latest payment of the user.
+        """
+        correct_order_payment_states = (
+            OrderPaymentState.BALANCE_DUE.value,
+            OrderPaymentState.FAILED.value,
+        )
+
+        recent_payment = kwargs.get('recent_payment')
+        order_payment_state = order_data['payment_state']
+
+        if order_payment_state not in correct_order_payment_states:
+            return None  # Order does not need draft payment. Halt pipeline.
+        elif not recent_payment:
+            return {}  # Continue. Will generate a new draft payment.
+        elif recent_payment['state'] == PaymentState.PENDING.value:
+            return None  # Order does not need draft payment. Halt pipeline.
+        elif recent_payment['state'] == PaymentState.FAILED.value:
+            return {}  # Continue. Will generate a new draft payment.
+        else:
```

Comment:
nit: we can skip `else:` keyword as there is already a `return` statement above it.

Reply:
Done in https://github.com/edx/commerce-coordinator/pull/108/commits/8882ad339cc40c1c9a821820997d2a1c5c67288a.